modules/ROOT/pages/bare-metal.adoc

```diff
@@ -165,3 +165,15 @@ Commonly bare metal systems will have a diversity of hardware - some systems may
 A useful approach is to script generating a per-machine `.iso`.  If you have a hardware database (whether a text file in git or relational database) then it will work to generate a per-machine `target-dell.ign` and `target-hp.ign` for example, and specify that to `--dest-ignition` alongside the appropriate `--dest-device` to generate `fedora-coreos-install-dell.iso` and `fedora-coreos-install-hp.iso`.
 
 Alternatively, instead of generating per-machine ISOs, you can have a `--pre-install` script run a privileged container which inspects the target system and writes an appropriate https://coreos.github.io/coreos-installer/customizing-install/#config-file-format[installer config] to `/etc/coreos/installer.d`.
+
+== Re-installing Fedora CoreOS
+
+You can use any of the methods described above to re-install Fedora CoreOS on the same machine via the live environment.
+
+=== Data persistence
+
+The installer does not completely scrub the target disk. Thanks to Ignition's https://coreos.github.io/ignition/operator-notes/#filesystem-reuse-semantics[filesystem-reuse semantics], if you apply a config with the exact same partition and filesystem settings as the first install, all previously stored data will still be available. This can be useful for persisting e.g. `/var/lib/containers` or `/var/home` between re-installs. See the xref:live-booting.adoc#_using_persistent_state[Using persistent state] section for more information.
```

Comment:
Due to the potential for data loss, we probably want to qualify this in a couple ways:

- This only works for data partitions, not for the rootfs.
- The data partitions must not specify `wipe_filesystem`, `wipe_volume`, or RAID volumes.

Those are mentioned in the linked doc, but we don't explicitly call out that there are limitations described at the other end of that link.  Also, the intro of that doc section describes a different goal than the one here.  So I'm wondering if we should do some restructuring: maybe we could pull out the common parts into a separate page/fragment and link/inline it in both places.

Should we also call out here that the root partition should be at least 8 GiB to prevent potentially clobbering data partitions on reinstall?

Reply:
We should also probably mention that it requires the keep partition logic for the rootfs disk?

Reply:
In general it doesn't.  That logic is mainly for the case where the data partitions were created outside Ignition, and (1) might vary from system to system or (2) can't be expressed in an Ignition config (i.e. aren't aligned to a 1 MiB boundary).

There is one caveat though, which is that the partition-saving logic will physically prevent the new OS image from clobbering the data partitions, even if the rootfs minimum-size rule is violated.  That's a nice property, but I'm still hesitant to recommend partition-saving instead of recreating the partitions in the Ignition config, since the former encourages multiple sources of truth.

In any event, we do need docs for the partition-saving functionality.

Reply:
> Due to the potential for data loss, we probably want to qualify this in a couple ways:
> 
>     * This only works for data partitions, not for the rootfs.
> 
>     * The data partitions must not specify `wipe_filesystem`, `wipe_volume`, or RAID volumes.
> 
> 
> Those are mentioned in the linked doc, but we don't explicitly call out that there are limitations described at the other end of that link.

Updated to mention this!

> Should we also call out here that the root partition should be at least 8 GiB to prevent potentially clobbering data partitions on reinstall?

Good call, done!